src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/correlation/CorrelationRule.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jmeter.protocol.http.correlation;
+
+import org.apache.jmeter.extractor.BoundaryExtractor;
+import org.apache.jmeter.extractor.HtmlExtractor;
+import org.apache.jmeter.extractor.RegexExtractor;
+import org.apache.jmeter.extractor.XPath2Extractor;
+import org.apache.jmeter.extractor.json.jsonpath.JSONPostProcessor;
+import org.apache.jmeter.protocol.http.gui.action.CorrelationRuleFile;
+
+import net.minidev.json.JSONObject;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+@JsonInclude(Include.NON_NULL)
+public final class CorrelationRule {
```

Comment:
Wouln't it be better to create mulitple classes inheriting from a base class ?

Reply:
Followed inheritance for the classes. Added more "Correlation Rule" classes like BoundaryExtractorRule with base class CorrelationRule.

Reply:
We have added the following new classes which inherit from CorrelationRule:
	-BoundryExtractorCorrelationRule
	-HtmlExtractorCorrelationRule
	-JsonExtractorCorrelationRule
	-RegexExtractorCorrelationRule
	-XPath2ExtractorCorrelationRule
This follows the inheritance. Please verify and check if the solution is as per your expectation.